vendor/k8s.io/kubernetes/openshift-kube-apiserver/admission/scheduler/podtolerationsrestriction/admission.go

```diff
@@ -0,0 +1,156 @@
+package podtolerationsrestriction
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/klog"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/admission/initializer"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	coreapi "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/serviceaccount"
+)
+
+const (
+	PluginName       = "scheduling.openshift.io/RestrictPodTolerations"
+	masterToleration = "node-role.kubernetes.io/master"
+)
+
+type podTolerations struct {
+	*admission.Handler
+	authorizer authorizer.Authorizer
+}
+
+var _ = initializer.WantsAuthorizer(&podTolerations{})
+var _ = admission.ValidationInterface(&podTolerations{})
+
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return NewPodTolerationsPlugin(), nil
+	})
+}
+
+func NewPodTolerationsPlugin() admission.Interface {
+	return &podTolerations{
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+	}
+}
+
+func (p *podTolerations) Validate(attr admission.Attributes, _ admission.ObjectInterfaces) error {
+	if attr.GetResource().GroupResource() != coreapi.Resource("pods") && attr.GetKind().GroupKind() != coreapi.Kind("Pod") && len(attr.GetSubresource()) == 0 {
```

Comment:
I'm not so great at parsing these.  Mind breaking them apart?

```go
if attr.groupResource != thing{
    return
}
if attr.groupkind != thing{
    return
}
if len() != 0{
    return
}
```

I think your conditional on len is backwards.

Reply:
Sure thing.